src/Security/Permission.php

```diff
@@ -293,33 +294,39 @@ public static function permissions_for_member($memberID)
         $groupList = Permission::groupList($memberID);
 
         if ($groupList) {
-            $groupCSV = implode(", ", $groupList);
-
-            $allowed = array_unique(
-                DB::withPrimary(fn() => DB::query("
-                    SELECT \"Code\"
-                    FROM \"Permission\"
-                    WHERE \"Type\" = " . Permission::GRANT_PERMISSION . " AND \"GroupID\" IN ($groupCSV)
-
-                    UNION
-
-                    SELECT \"Code\"
-                    FROM \"PermissionRoleCode\" PRC
-                    INNER JOIN \"PermissionRole\" PR ON PRC.\"RoleID\" = PR.\"ID\"
-                    INNER JOIN \"Group_Roles\" GR ON GR.\"PermissionRoleID\" = PR.\"ID\"
-                    WHERE \"GroupID\" IN ($groupCSV)
-                "))->column() ?? []
-            );
-
-            $denied = array_unique(
-                DB::withPrimary(fn() => DB::query("
-                    SELECT \"Code\"
-                    FROM \"Permission\"
-                    WHERE \"Type\" = " . Permission::DENY_PERMISSION . " AND \"GroupID\" IN ($groupCSV)
-                "))->column() ?? []
-            );
-
-            return array_diff($allowed ?? [], $denied);
+            // Get the allowed permissions for this user based on the permissions assigned
+            // directly to its groups, and the permissions assigned to roles on its groups.
+            $allowed = Permission::get()
```

Comment:
A few changes here:

1. Using the ORM instead of raw SQL queries
2. Using dynamic table names instead of hardcoded ones
3. Reducing total number of queries from 2 to 1

Reply:
This logic is covered by `PermissionTest::testPermissionsForMember()`